### PR TITLE
Issue #2054 Total memory regex fails in cmd-config.feature

### PR DESCRIPTION
**Pull request by agajdosi**

### CODE REVIEW

**Comment by agajdosi** [comment]
Hi @amitkrout, please validate this fix on the machine on which have faced the issue.

File: test/integration/features/cmd-config.feature [hunk 1]
```diff
@@ -213,7 +213,7 @@ user defined options which changes default behaviour of Minishift.
 
   Scenario: Checking that memory value was applied
      When executing "minishift ssh -- cat /proc/meminfo" succeeds
-     Then stdout should match "MemTotal:\s*3[4-5][0-9]{5} kB"
+     Then stdout should match "MemTotal:\s*3[4-7][0-9]{5} kB"
```
Comment (budhrg) [inline test/integration/features/cmd-config.feature]: This can be `[4-9]` right? How can we sure `[4-7]` is correct here?

Comment (budhrg) [reply test/integration/features/cmd-config.feature]: Now we have agreed on `MemTotal:\s*3[0-9]{6} kB`

File: test/integration/features/cmd-config.feature [hunk 2]
```diff
@@ -213,7 +213,7 @@ user defined options which changes default behaviour of Minishift.
 
   Scenario: Checking that memory value was applied
      When executing "minishift ssh -- cat /proc/meminfo" succeeds
-     Then stdout should match "MemTotal:\s*3[4-5][0-9]{5} kB"
+     Then stdout should match "MemTotal:\s*3[4-9][0-9]{5} kB"
```
Comment (budhrg) [inline test/integration/features/cmd-config.feature]: @agajdosi , I think we agreed on `Then stdout should match "MemTotal:\s*3[0-9]{6} kB"`.

Comment (budhrg) [reply test/integration/features/cmd-config.feature]: As per comment here https://issues.jboss.org/browse/CDK-256?focusedCommentId=13537041&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13537041, `MemTotal` can start with 3 or reach to 5.
 
@amitkrout @agajdosi , I think we should not worry about `3` or `4` or `5` here. It is hypervisor behavior. 

Let's go with `Then stdout should match "MemTotal:\s*\d[0-9]{6} kB"`

Comment (amitkrout) [reply test/integration/features/cmd-config.feature]: @budhrg Right, We have no control over issue which is coming from hypervisor side. I am ok with this patch

Comment (agajdosi) [reply test/integration/features/cmd-config.feature]: @budhrg @amitkrout then I think we need to cover also a situation with 4[0-1]**** kbs as this sometimes happens on Windows 10.

> The pass rate is 60% with the patch (failed 2 out of 5 run). In one of the test run i can see memory increase to 41* on applied 3500mb. Seems the hypervisor is responsible for inconsistent in "MemTotal" output on same applied memory in config file.

**Comment by agajdosi** [comment]
Hi @LalatenduMohanty @amitkrout, expected value for 3500mb, should be around `3.584.000kb`, I have modified the regexp to match interval of 3.300.000 to 3.699.999 kb. Bigger deviation we should not tolerate IMHO. For the Hyper-V bug I think we can document it and ignore it in the nightly tests once the hyper-V slave is ready again.

Other take might be to disable the test completely. wdyt?

**Comment by gbraad** [comment]
I think you could test for a value that is at least bigger than a certain
threshold. This feels less likely to fail... As we are not test h the
supervisor drivers,  it just if the setting has been applied and worked in
some way

-- 

   Gerard Braad | http://gbraad.nl
   [ Doing Open Source Matters ]

**Comment by amitkrout** [comment]
Have some observation with these config set memory values ```4096``` and ```4GB```. This is what i am seeing the respective value applied in ```/proc/meminfo```

```
Config set memory : 4096        Applied memory : 4299584 kB
Config set memory : 4GB         Applied memory : 4029248 kB
```
Seems hyperv considerable increasing the the memory value are passed in mbs with respect to the same memory in GBs.  Instead of passing memory ```3500```, set the memory with 3GB probably solve the problem IMO.

**Comment by agajdosi** [comment]
@gbraad I have now checked on the Win 10 (using ) by myself and it seems, that in some cases the value is taken properly - the hyperV shows 3500mb and /proc/meminfo shows 3.400.000 kb, however in some cases the HyperV shows 4096mb and /proc/meminfo shows 4.060.000 kb.

I did just few runs (had to borrow the machine to Budh then) but I realized that the fialing starts are probably those, which are run directly after setup-cdk: `setup-cdk; config set memory 3500; start` fails, while after that when I delete the machine and start again, the proper values are applied.

Will investigate more.

File: test/integration/features/cmd-config.feature [hunk 3]
```diff
@@ -211,10 +211,6 @@ user defined options which changes default behaviour of Minishift.
      Then stdout should match "vCPUs\s*:\s*3"
       And Minishift should have state "Running"
 
-  Scenario: Checking that memory value was applied
-     When executing "minishift ssh -- cat /proc/meminfo" succeeds
-     Then stdout should match "MemTotal:\s*3[4-5][0-9]{5} kB"
-
```
Comment (praveenkumar) [inline test/integration/features/cmd-config.feature]: As per yesterday discussion, this looks good 👍

**Comment by amitkrout** [comment]
No other reference found for ```minishift ssh -- cat /proc/meminfo``` in remaining feature file. So LGTM :+1:

**Pull request merged**